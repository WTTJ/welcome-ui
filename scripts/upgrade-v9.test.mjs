import fs from 'fs'
import os from 'os'
import path from 'path'

import { describe, expect, it } from 'vitest'

import * as upgradeScript from './upgrade-v9.mjs'

describe('upgrade-v9 migration script', () => {
  describe('processComponents (integration)', () => {
    it('migrates fixtures/Component.tsx as expected', async () => {
      // Setup temp dir and copy fixture
      const tmpDir = fs.mkdtempSync(path.join(os.tmpdir(), 'wui-upgrade-test-'))
      const srcFile = path.join(__dirname, 'fixtures', 'Component.tsx')
      const destFile = path.join(tmpDir, 'Component.tsx')
      fs.copyFileSync(srcFile, destFile)

      // Find components in the temp file
      const components = upgradeScript.findBoxComponents(tmpDir)
      // Run migration (shouldReplace = true)
      await upgradeScript.processComponents(components, true)

      // Read migrated file
      const migrated = fs.readFileSync(destFile, 'utf8')

      // Assert output (snapshot or inline string)
      expect(migrated).toMatchSnapshot()
    })
  })

  describe('findBoxComponents', () => {
    it('finds Box, Flex, Grid, Stack components in a directory', () => {
      // Setup: create a mock directory structure in memory or use a temp dir
      // For demonstration, just check that the function runs and returns an array
      const result = upgradeScript.findBoxComponents(path.join(__dirname, 'fixtures'))
      expect(Array.isArray(result)).toBe(true)
    })
  })

  describe('transformValue', () => {
    it('transforms known keys using valueMap', () => {
      expect(upgradeScript.transformValue('align', 'center')).toContain('items')
      expect(upgradeScript.transformValue('backgroundColor', 'red')).toContain('bg')
    })
    it('returns undefined for unknown keys', () => {
      expect(upgradeScript.transformValue('unknownKey', 'foo')).toBeUndefined()
    })
  })

  describe('transform', () => {
    it('handles 100% as full', () => {
      expect(upgradeScript.transform('w', '100%')).toBe('w-full')
    })
    it('handles numeric values', () => {
      expect(upgradeScript.transform('w', '16')).toContain('w-[')
    })
    it('handles 0 as zero class', () => {
      expect(upgradeScript.transform('m', '0')).toBe('m-0')
    })
  })

  describe('transformSpecificValue', () => {
    it('handles grid-cols special cases', () => {
      expect(upgradeScript.transformSpecificValue('grid-cols-1fr')).toBe('grid-cols-1')
      expect(upgradeScript.transformSpecificValue('grid-cols-1fr 1fr')).toBe('grid-cols-2')
    })
    it('returns value for non-grid-cols', () => {
      expect(upgradeScript.transformSpecificValue('foo')).toBe('foo')
    })
  })
})
