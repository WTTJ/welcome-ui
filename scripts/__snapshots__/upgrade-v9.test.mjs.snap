// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`upgrade-v9 migration script > processComponents (integration) > migrates fixtures/Component.tsx as expected 1`] = `
"import { Alert } from '../../lib/src/old/components/Alert'
import { Box } from '../../lib/src/old/components/Box'
import { IconsFont } from '../../lib/src/old/components/IconsFont'
import { Link } from '../../lib/src/old/components/Link'
import { Text } from '../../lib/src/old/components/Text'

const FormattedMessage = ({
  defaultMessage,
  id,
}: {
  defaultMessage: string
  id: string
}): React.ReactNode => <span id={id}>{defaultMessage}</span>

const signinPath = '/signin'

const LAYOUT_WIDTH = {
  normal: 400,
  terms: 560,
}

// Usage examples
export const Example = () => (
  <>
    <div className="flex h-full">
      <aside className="bg-beige-20 flex-col h-max-content items-center justify-center max-w-[30rem] md:flex none min-h-full w-40%">
        <div className="pt-3xl">
          <Link to={signinPath}>
            <IconsFont.Add h={58} />
          </Link>
        </div>
        <div className="flex flex-col grow items-center justify-center lg:px-3xl md:px-xl max-w-[26.25rem] pb-xl w-full">
          <Text as="h2" mb="xxl" mt="0" variant="h3">
            This is my content
          </Text>
        </div>
      </aside>
      <div className="flex flex-[0.0625rem] flex-col">
        <div className="grid grow items-center max-w-LAYOUT_WIDTH['normal'] px-lg w-full" mx="auto">
          <FormattedMessage defaultMessage="Terms of use" id="EBDrtq" />
        </div>
        <div className="md:none mt-xl">
          <FormattedMessage defaultMessage="Terms of use" id="EBDrtq" />
        </div>
      </div>
    </div>
    <Alert className="mb-xl" isFullWidth variant="danger">
      <Alert.Title>
        <FormattedMessage defaultMessage="Please sign in with provider" id="5nmpcx" />
      </Alert.Title>
      <span>
        <FormattedMessage
          defaultMessage="Your organization enforces provider sign-in. Contact your administrator if you need help."
          id="TzKv+W"
        />
      </span>
    </Alert>
  </>
)
"
`;
